addons/binding/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/internal/net/HttpRequest.java

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.neeo.internal.net;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.Invocation.Builder;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.apache.http.HttpStatus;
+import org.openhab.binding.neeo.NeeoUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class represents an HTTP session with a client
+ *
+ * @author Tim Roberts - Initial contribution
+ */
+public class HttpRequest implements AutoCloseable {
```

Comment:
I do this because I can customize everything directly in this class rather than across code.  Prior version of NEEO had to have customized headers that was simple with these classes (they have dropped those since then) but I wanted to keep them just in case something get's reintroduced (since all the API I'm using is not really public per se and does tend to change)

Reply:
@martinvw 
Please remember - the firmware is on .45 and we've had about 10 releases *since* the initial review.  NEEO is sort of a moving target and I've keep alot of flexibility in to move easily with it.  Much of that flexibility has not been used (yet) or was used in the past but doesn't apply to the current firmware.  This is one of those cases - a prior firmware needed custom headers but the current one doesn't.  There is nothing to say they won't reintroduce that again in a future firmware - so I'd rather keep this to easily reintroduce that type of logic rather than have to reinvent the code (and everything that calls it) if that situation happens again.